Source/WebCore/dom/Node.cpp

```diff
@@ -2102,6 +2106,21 @@ Element* Node::enclosingLinkEventParentOrSelf()
     return nullptr;
 }
 
+void Node::expandAnyDetailsAncestor()
+{
+    if (!document().settings().detailsAutoExpandEnabled())
+        return;
+    ShouldNotFireMutationEventsScope scope(document());
+    for (auto& element : ancestorsOfType<HTMLElement>(*this)) {
+        if (is<HTMLSummaryElement>(element))
+            continue;
+        if (is<HTMLDetailsElement>(element) && !element.hasAttributeWithoutSynchronization(HTMLNames::openAttr)) {
+            downcast<HTMLDetailsElement>(element).toggleOpen();
```

Comment:
```suggestion
        if (RefPtr detailsElement = dynamicDowncast<HTMLDetailsElement>(element)) {
            if (!element.hasAttributeWithoutSynchronization(HTMLNames::openAttr)) {
                detailsElement->toggleOpen();
```

Can you use dynamicDowncast?

Reply:
> Can you use dynamicDowncast?

Yup, thanks — made it so